src/timeoffset.rs

```diff
@@ -0,0 +1,362 @@
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(t_ref_nanos, TimeScale::GST, TimeScale::GPST, polynomials)
+            .unwrap()
+    }
+
+    /// Define a new |[TimeScale::BDT] - [TimeScale::GPST]| [TimeOffset] from time of week components
+    pub fn from_bdt_gpst_time_of_week(
+        t_ref_nanos: (u32, u64),
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(t_ref_nanos, TimeScale::BDT, TimeScale::GPST, polynomials)
+            .unwrap()
+    }
+
+    /// Define a new |[TimeScale::BDT] - [TimeScale::UTC]| [TimeOffset] from time of week components
+    pub fn from_bdt_utc_time_of_week(
+        t_ref_nanos: (u32, u64),
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(t_ref_nanos, TimeScale::BDT, TimeScale::UTC, polynomials)
+            .unwrap()
+    }
+
+    /// Define a new |[TimeScale::BDT] - [TimeScale::GST]| [TimeOffset] from time of week components
+    pub fn from_bdt_gst_time_of_week(
+        t_ref_nanos: (u32, u64),
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(t_ref_nanos, TimeScale::BDT, TimeScale::GST, polynomials)
+            .unwrap()
+    }
+
+    /// Define a new |[TimeScale::QZSST] - [TimeScale::GPST]| [TimeOffset] from time of week components
+    pub fn from_qzsst_gpst_time_of_week(
+        t_ref_nanos: (u32, u64),
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(
+            t_ref_nanos,
+            TimeScale::QZSST,
+            TimeScale::GPST,
+            polynomials,
+        )
+        .unwrap()
+    }
+
+    /// Define a new  |[TimeScale::QZSST] - [TimeScale::UTC]|  [TimeOffset] from time of week components
+    pub fn from_qzsst_utc_time_of_week(
+        t_ref_nanos: (u32, u64),
+        polynomials: (f64, f64, f64),
+    ) -> Self {
+        Self::from_reference_time_of_week(
+            t_ref_nanos,
+            TimeScale::QZSST,
+            TimeScale::UTC,
+            polynomials,
+        )
+        .unwrap()
+    }
+
+    /// Returns both [TimeScale]s this [TimeOffset] allows converting to.
+    pub fn supported_timescales(&self) -> (TimeScale, TimeScale) {
+        (self.lhs, self.rhs)
+    }
+
+    /// Update this [TimeOffset] with new reference epoch and polynomials.
+    /// NB: this should be expressed in the left-hand side [TimeScale] and we have no means
+    /// to verify that.
+    pub fn update_mut(&mut self, t_ref_nanos: (u32, u64), polynomials: (f64, f64, f64)) {
+        self.t_ref_nanos = t_ref_nanos;
+        self.polynomials = polynomials;
+    }
+
+    /// Define a new [TimeOffset] with new reference [TimeScale], while preserving other components.
+    /// NB: this should be expressed in the left-hand side [TimeScale] and we do not verify it!
+    pub fn with_reference_timescale(&self, ts: TimeScale) -> Self {
```

Comment:
```suggestion
    pub fn with_reference_timescale(mut self, ts: TimeScale) -> Self {
        self.rhs = ts;
        self
    }
```

I learned only recently that for `Copy` structs, you can just move the structure with the function call, and avoid calling `clone`.

Reply:
I tried to apply this everywhere. Does that buy you a memcopy ?